packages/admin/admin/src/mui/menu/Menu.tsx

```diff
@@ -53,54 +139,50 @@ const MenuDrawer: React.FC<WithStyles<typeof styles> & MenuProps> = ({
         });
     }, [history, variant, open, toggleOpen]);
 
-    const temporaryDrawerClasses: string[] = [classes.drawer, classes.temporary];
-    temporaryDrawerClasses.push(open ? classes.open : classes.closed);
-
-    const permanentDrawerClasses: string[] = [classes.drawer, classes.permanent];
-    permanentDrawerClasses.push(open ? classes.open : classes.closed);
-
     const temporaryOpen = variant === "temporary" && open;
     const permanentOpen = variant === "permanent" && open;
 
     // Always render both temporary and permanent drawers to make sure, the opening and closing animations run fully when switching between variants.
     return (
         <>
-            <Drawer
+            <TemporaryDrawer
                 variant="temporary"
-                className={temporaryDrawerClasses.join(" ")}
+                {...slotProps?.temporaryDrawer}
                 // workaround for issue: https://github.com/mui/material-ui/issues/35793
                 open={initialRender.current ? false : temporaryOpen}
-                PaperProps={{ style: { width: drawerWidth }, ...temporaryDrawerPaperProps }}
+                ownerState={ownerState}
+                PaperProps={{
+                    ...temporaryDrawerPaperProps,
+                    sx: { width: drawerWidth, ...temporaryDrawerPaperProps?.sx },
+                }}
                 onClose={toggleOpen}
-                {...temporaryDrawerProps}
+                {...restProps}
             >
                 {children}
-            </Drawer>
+            </TemporaryDrawer>
 
-            <Drawer
+            <PermanentDrawer
                 variant="permanent"
-                className={permanentDrawerClasses.join(" ")}
+                {...slotProps?.permanentDrawer}
                 open={permanentOpen}
-                style={{ width: permanentOpen ? drawerWidth : 0 }}
+                ownerState={ownerState}
                 PaperProps={{
                     elevation: 2,
-                    style: {
+                    ...permanentDrawerPaperProps,
+                    sx: {
                         top: headerHeight,
                         height: `calc(100% - ${headerHeight}px)`,
                         width: drawerWidth,
                         marginLeft: permanentOpen ? 0 : -drawerWidth,
+                        ...permanentDrawerPaperProps?.sx,
                     },
-                    ...permanentDrawerPaperProps,
                 }}
-                {...permanentDrawerProps}
             >
```

Comment:
In this case the `restProps` should probably be applied to both Drawers, as there is no real "Root" element, but both drawers are sort of the primary element in this component, depending on which one is shown. 

```suggestion
                {...restProps}
            >
```